backend/database.py

```diff
@@ -95,6 +95,9 @@ def select(self, query: str, parameters: Iterable = None, fetch_all: bool = True
         cursor = self.__immediate_connection.cursor(dictionary=True, buffered=True)
         cursor.execute(query, parameters)
 
+        if cursor.rowcount == 0:
+            return None
```

Comment:
Will the other end throw a NoneType-related exception?

Reply:
shit yeah. An empty dict would be better and reflect this more.